src/main/java/seedu/address/logic/commands/CommandResult.java

```diff
@@ -22,28 +25,55 @@ public class CommandResult {
     /**
      * Constructs a {@code CommandResult} with the specified fields.
      */
-    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit) {
+    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit,
+                         boolean isDisplayCommand, Person firstMatchedPerson) {
+
         this.feedbackToUser = requireNonNull(feedbackToUser);
         this.showHelp = showHelp;
         this.exit = exit;
+        this.isDisplayCommand = isDisplayCommand;
+        this.firstMatchedPerson = firstMatchedPerson;
+    }
+    /**
+     * Constructs a CommandResult object with the specified feedback to the user and the first matched person.
+     *
+     * @param feedbackToUser The feedback message to be displayed to the user.
+     * @param firstMatchedPerson The first matched person associated with the command result.
+     * @throws NullPointerException If {@code firstMatchedPerson} is null.
+     */
+    public CommandResult(String feedbackToUser, Person firstMatchedPerson) {
+        this(feedbackToUser, false, false, true, firstMatchedPerson);
+        requireNonNull(firstMatchedPerson);
+    }
+
+    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit) {
+        this(feedbackToUser, showHelp, exit, false, null);
     }
 
     /**
      * Constructs a {@code CommandResult} with the specified {@code feedbackToUser},
      * and other fields set to their default value.
      */
     public CommandResult(String feedbackToUser) {
-        this(feedbackToUser, false, false);
+        this(feedbackToUser, false, false, false, null);
     }
 
     public String getFeedbackToUser() {
         return feedbackToUser;
     }
 
+    public Person getPerson() {
+        return firstMatchedPerson;
+    }
+
     public boolean isShowHelp() {
         return showHelp;
```

Comment:
Refer to this naming convention: https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/topics.html#w4-6-code-quality-naming for boolean variables.
Might be required to change showHelp to isShowHelp